### DISCUSSION TITLE
Building Wealthfolio Roadmap

### DISCUSSION THREAD

**Discussion by afadil** [afadil/wealthfolio#70] [Ideas]
Here is a draft of my vision to this project. Any ideas are welcome:

## Vision
To help individuals build their wealth by providing a safe, privacy-focused and easy-to-use tool to track their investments and goals.

## Strategy
1. Build a polished application that focuses on the essentials
2. Underdo the competition by offering a simpler, more focused alternative
3. Fight feature creep and resist creating a bloated app
4. Avoid complexity at all costs
5. Provide clear insights on how your portfolio is performing against your goals

  ## What We Believe In
  1. Your Privacy: Your money info stays on your computer. You retain complete control over your financial information..
  2. Easy to Use: Anyone can track their investments with our simple design.
  3. No Hidden Costs: Free or Pay once to support. Use forever—no surprise fees or subscriptions.
  4. Lots of Features: Track many types of investments and use helpful tools.
  5. Beautiful, Focused, and Boring App: We believe in crafting an app that's visually appealing yet doesn't distract from its core purpose.

### Step 1: The Basics (Now)
- [x] Track your investments
- [x] Store data on your computer
- [x] Simple, easy-to-use design

### Step 2: More Features (Next 3-6 months)
- [x] Let you export your data (CSV, sqlite file, json)
- [x] - Track contributions limits 
- [x] Enhance portfolio History calculation
- [x] Support Stock Splitting
- [ ] Advanced CSV import 
- [ ] Support another market data API (Bring your own API keys) (example https://www.alphavantage.co)
- [ ] Let you track more things like real estate, commodities, and other assets
- [ ] Build install packages for the remaining platforms

### Step 3: Smart Tools (6-12 months)
- [ ] Some advanced portfolio analysis and insights
- [ ] Add Monte-Carlo projection
- [ ] Support another market data API

### Step 4: Growing (12-18 months)
- [ ] Make a mobile app that works with your phone securely
- [ ] Connect with some brokers/ plateform to fetch and update activities automatically

### Step 5: Building a Community (18+ months)
- [ ] Add support for Options trading

### Always Doing
- Keeping the app safe and up-to-date
- Making the app faster and bette using user feedback

We're building Wealthfolio to be a great alternative to online investment trackers. Our plan might change based on what users tell us they want, but we'll always focus on keeping your data private and helping you manage your money better.

**Comment by afadil** [comment]
@maidh91 @alexanderameye here is draft of the vision and roadmap of this project. I really appreciate your insights and contributions.

**Comment by rsalus** [comment]
@afadil What did you have in mind for the advanced CSV import feature? I'd love to be able to automate importing my CSV files into the app. Ideally, I'd be able to have a workflow something like this:

-  I'd likely be running Wealthfolio in a VM or container.
-  Some automated process (script/function) dumps CSVs into a shared directory.
-  Wealthfolio detects and imports the CSVs.

Of course, there would have to be some way for the import process to be triggered. Maybe a polling or event system? What are your thoughts?

**Comment by fouadchamoun** [comment]
@afadil thank you for this beautiful app. I cannot overstate how much more polished and well thought out it feels compared to other apps I've been benchmarking for my personal use. 

a couple of thoughts I had after my first steps in the app:

- i need to be sure my data isn't lost if my Mac fails. maybe iCloud sync ? or using iCloud Drive as storage instead of the local system folder. When selfhosting an app, I'm used to running it in docker and backup up the data volume. It's been a while since i've used a desktop app 🙂
- i expected an option for "reinvested" dividends but found that they only count as cash. to me this feels like a essential feature for dividend tracking. have you had any other similar feedback ?

thanks again. i hope my feedback helps. and I hope to even be able to contribute to the development one day.
